web/api/helpers/verify.ts

```diff
@@ -0,0 +1,284 @@
+  } catch (error) {
+    logger.error("Error create signal hash", { error });
+    return {
+      error: {
+        message:
+          "This attribute is improperly formatted. Expected an ABI-encoded uint256 or a string.",
+        code: "invalid_format",
+        statusCode: 400,
+        attribute: "signal",
+      },
+    };
+  }
+
+  return {
+    params: {
+      proof,
+      nullifier_hash,
+      external_nullifier,
+      signal_hash,
+      merkle_root,
+    },
+  };
+};
+
+/**
+ * Verifies a ZKP with the World ID smart contract
+ */
+export const verifyProof = async (
+  proofParams: IInputParams,
+  verifyParams: IVerifyParams,
+): Promise<{
+  success?: true;
+  status?: string;
+  error?: IInternalError;
+}> => {
+  // Parse the inputs
+  const parsed = parseProofInputs(proofParams);
+  if (parsed.error || !parsed.params) {
+    return { error: parsed.error };
+  }
+
+  const { params: parsedParams } = parsed;
+
+  // Query the signup sequencer to verify the proof
+  const body = JSON.stringify({
+    root: parsedParams.merkle_root,
+    nullifierHash: parsedParams.nullifier_hash,
+    externalNullifierHash: parsedParams.external_nullifier,
+    signalHash: parsedParams.signal_hash,
+    proof: parsedParams.proof,
+  });
+
+  const sequencerUrl =
+    sequencerMapping[verifyParams.verification_level]?.[
+      verifyParams.is_staging.toString()
+    ];
+
+  const response = await fetch(
+    verifyParams.max_age
+      ? `${sequencerUrl}/verifySemaphoreProof?maxRootAgeSeconds=${verifyParams.max_age}`
+      : `${sequencerUrl}/verifySemaphoreProof`,
+    {
+      method: "POST",
+      headers: {
+        "Content-Type": "application/json",
+      },
+      body,
+    },
+  );
+
+  if (!response.ok) {
+    try {
+      const rawErrorMessage = await response.text();
+      const knownError = KNOWN_ERROR_CODES.find(
+        ({ rawMessage }) => rawMessage === rawErrorMessage,
+      );
+      return {
+        error: {
+          message:
+            knownError?.detail ||
+            `We couldn't verify the provided proof (error code ${rawErrorMessage}).`,
+          code: knownError?.code || "invalid_proof",
+          statusCode: 400,
+          attribute: null,
+        },
+      };
+    } catch {
+      return {
+        error: {
+          message: "There was an internal issue verifying this proof.",
+          code: "internal_error",
+          statusCode: 500,
+        },
+      };
+    }
+  }
+
+  const result = await response.json();
+  const status = result.status === "mined" ? "on-chain" : "pending";
+
+  if (!status) {
+    logger.error("Unexpected response received from sequencer.", {
+      result,
+      sequencerUrl,
+    });
+    throw new Error("Unexpected response received from sequencer.");
+  }
+
+  return { success: true, status };
+};
+
+/**
+ * Checks whether the person can be verified for a particular action based on the max number of verifications
+ */
+export const canVerifyForAction = (
```

Comment:
what's the reason for not deleting the other canVerifyForAction inside of api/helpers? Looks like the undefined check can be removed

Reply:
removed the undefined check

we could use this definition of canVerifyForAction for the v1/verify endpoint as well, but i think it's best to keep the v1 endpoint code/helpers in the legacy folder to prevent any unintended side effects of making changes for the v2 endpoints